[Bug] Deepseek v4 :torch._inductor.exc.InductorError: AssertionError:

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>


### 🐛 Describe the bug

When launching deepseek v4 with tp8 on H100, encountered the following error:

```
WorkerProc hit an exception.
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962] Traceback (most recent call last):
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/executor/multiproc_executor.py", line 957, in worker_busy_loop
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     output = func(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]              ^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/utils/_contextlib.py", line 124, in decorate_context
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return func(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/worker/gpu_worker.py", line 370, in determine_available_memory
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     self.model_runner.profile_run()
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/worker/gpu_model_runner.py", line 5848, in profile_run
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     hidden_states, last_hidden_states = self._dummy_run(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                                         ^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/utils/_contextlib.py", line 124, in decorate_context
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return func(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/worker/gpu_model_runner.py", line 5537, in _dummy_run
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     outputs = self.model(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]               ^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/cuda_graph.py", line 254, in __call__
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return self.runnable(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1779, in _wrapped_call_impl
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return self._call_impl(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1790, in _call_impl
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return forward_call(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/deepseek_v4.py", line 1474, in forward
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     hidden_states = self.model(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                     ^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/decorators.py", line 623, in __call__
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     self.aot_compiled_fn = self.aot_compile(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/wrapper.py", line 183, in aot_compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return self._compiled_callable.aot_compile((args, kwargs))
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_dynamo/eval_frame.py", line 873, in aot_compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return aot_compile_fullgraph(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_dynamo/aot_compile.py", line 368, in aot_compile_fullgraph
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     compiled_fn = backend(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                   ^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/__init__.py", line 2535, in __call__
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return self.compiler_fn(model_, inputs_, **self.kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/lib/python3.12/contextlib.py", line 81, in inner
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return func(*args, **kwds)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/backends.py", line 1194, in __call__
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     PiecewiseCompileInterpreter(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/tracing/otel.py", line 178, in sync_wrapper
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return func(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/backends.py", line 721, in run
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return super().run(*args)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/fx/interpreter.py", line 200, in run
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     self.env[node] = self.run_node(node)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                      ^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/fx/interpreter.py", line 297, in run_node
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return getattr(self, n.op)(n.target, args, kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/backends.py", line 748, in call_module
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     piecewise_backend = PiecewiseBackend(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                         ^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/piecewise_backend.py", line 190, in __init__
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     self.compile_all_ranges()
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/piecewise_backend.py", line 266, in compile_all_ranges
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     range_entry.runnable = self.vllm_backend.compiler_manager.compile(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/tracing/otel.py", line 178, in sync_wrapper
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return func(*args, **kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/backends.py", line 351, in compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     compiled_graph, handle = self.compiler.compile(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                              ^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/vllm/compilation/compiler_interface.py", line 372, in compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     compiled_graph = standalone_compile(graph, example_inputs, **compile_kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/__init__.py", line 444, in standalone_compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return standalone_compile(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/standalone_compile.py", line 444, in standalone_compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     compiled_fn = compile_fx(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                   ^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 2527, in compile_fx
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return compile_fx(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 2578, in compile_fx
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return _maybe_wrap_and_compile_fx_main(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 2655, in _maybe_wrap_and_compile_fx_main
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return _compile_fx_main(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 2864, in _compile_fx_main
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     raise e.remove_dynamo_frames() from None  # see TORCHDYNAMO_VERBOSE=1
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 1053, in _compile_fx_inner
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     raise InductorError(e, currentframe()).with_traceback(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 1037, in _compile_fx_inner
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     mb_compiled_graph = fx_codegen_and_compile(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]                         ^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 1798, in fx_codegen_and_compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return scheme.codegen_and_compile(gm, example_inputs, inputs_to_check, graph_kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 1344, in codegen_and_compile
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     _recursive_post_grad_passes(gm, is_inference=is_inference)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 583, in _recursive_post_grad_passes
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     post_grad_passes(gm, is_inference)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/fx_passes/post_grad.py", line 357, in post_grad_passes
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     ).apply_graph_pass(decompose_triton_kernel_wrapper_functional)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/fx/passes/graph_transform_observer.py", line 103, in apply_graph_pass
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     return pass_fn(self.gm.graph)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/fx_passes/post_grad.py", line 1255, in decompose_triton_kernel_wrapper_functional
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     graph_pass.apply(graph)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/pattern_matcher.py", line 2063, in apply
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     entry.apply(m, graph, node)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/pattern_matcher.py", line 1132, in apply
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     self.handler(match, *match.args, **match.kwargs)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/fx_passes/post_grad.py", line 1253, in _
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     match.replace_by_example(decomp, flat_args, run_functional_passes=False)
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/pattern_matcher.py", line 316, in replace_by_example
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]     assert len(graph_with_eager_vals.graph.nodes) == len(
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker_TP6_EP6 pid=13431) ERROR 04-28 06:25:58 [multiproc_executor.py:962] torch._inductor.exc.InductorError: AssertionError: 
```

launch command:
```
vllm serve /mnt/model/DeepSeek-V4-Flash   --trust-remote-code   --kv-cache-dtype fp8   --block-size 256   --enable-expert-parallel   --data-parallel-size 1 --tensor-parallel-size 8   --compilation-config '{"cudagraph_mode":"FULL_AND_PIECEWISE", "custom_ops":["all"]}'   --max-num-batched-tokens 8192   --max-model-len auto   --max-num-seqs 128   --gpu-memory-utilization 0.95   --reasoning-parser deepseek_v4 --port 7888
```

image env:
`vllm/vllm-openai:v0.20.0-cu130`


### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/vllm/v1/attention/ops/deepseek_v4_ops/fused_inv_rope_fp8_quant.py
+++ b/vllm/v1/attention/ops/deepseek_v4_ops/fused_inv_rope_fp8_quant.py
@@ -10,6 +10,7 @@ INT32-packed UE8M0 on SM100) so fp8_einsum skips transform_sf_into_required_layo
 import torch
 
 from vllm.triton_utils import tl, triton
+from vllm.utils.torch_utils import direct_register_custom_op
 
 
 @triton.jit
@@ -180,34 +181,74 @@ def fused_inv_rope_fp8_quant(
     fp8_dtype = torch.float8_e4m3fn
     fp8_max = torch.finfo(fp8_dtype).max
 
-    fp8_buf = torch.empty(
-        (n_groups, num_tokens, d),
-        dtype=fp8_dtype,
-        device=o.device,
-    )
-
     tma_aligned_T = get_tma_aligned_size(num_tokens, 4)
     if tma_aligned_scales:
         packed_sf_k = (num_scale_blocks + 3) // 4
-        scale_buf = torch.empty(
-            n_groups * packed_sf_k * tma_aligned_T,
-            dtype=torch.int32,
-            device=o.device,
-        ).as_strided(
-            (n_groups, num_tokens, packed_sf_k),
-            (packed_sf_k * tma_aligned_T, 1, tma_aligned_T),
-        )
+        scale_inner = packed_sf_k
     else:
-        scale_buf = torch.empty(
-            n_groups * num_scale_blocks * tma_aligned_T,
-            dtype=torch.float32,
-            device=o.device,
-        ).as_strided(
-            (n_groups, num_tokens, num_scale_blocks),
-            (num_scale_blocks * tma_aligned_T, 1, tma_aligned_T),
-        )
+        scale_inner = num_scale_blocks
 
-    common_args = dict(
+    # Run kernel through a custom op so inductor sees an opaque boundary.
+    # It's a pytorch bug, see https://github.com/vllm-project/vllm/issues/41106
+    fp8_buf, scale_buf = torch.ops.vllm.fused_inv_rope_fp8_quant_kernel(
+        o,
+        positions,
+        cos_sin_cache,
+        heads_per_group,
+        quant_group_size,
+        chunks_per_head,
+        nope_dim % quant_group_size,
+        rope_dim // 2,
+        tma_aligned_scales,
+        fp8_max,
+        tma_aligned_T,
+        num_tokens,
+        n_groups,
+        d,
+        scale_inner,
+    )
+    return fp8_buf.transpose(0, 1), scale_buf.transpose(0, 1)
+
+
+def _fused_inv_rope_fp8_quant_kernel_impl(
+    o: torch.Tensor,
+    positions: torch.Tensor,
+    cos_sin_cache: torch.Tensor,
+    heads_per_group: int,
+    quant_group_size: int,
+    chunks_per_head: int,
+    rope_start: int,
+    half_rope: int,
+    tma_aligned_scales: bool,
+    fp8_max: float,
+    tma_aligned_T: int,
+    num_tokens: int,
+    n_groups: int,
+    d: int,
+    scale_inner: int,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    fp8_buf = torch.empty(
+        (n_groups, num_tokens, d),
+        dtype=torch.float8_e4m3fn,
+        device=o.device,
+    )
+    scale_dtype = torch.int32 if tma_aligned_scales else torch.float32
+    scale_buf = torch.empty(
+        n_groups * scale_inner * tma_aligned_T,
+        dtype=scale_dtype,
+        device=o.device,
+    ).as_strided(
+        (n_groups, num_tokens, scale_inner),
+        (scale_inner * tma_aligned_T, 1, tma_aligned_T),
+    )
+    grid = (tma_aligned_T, n_groups * heads_per_group)
+    _fused_inv_rope_fp8_quant_per_head[grid](
+        o,
+        positions,
+        cos_sin_cache,
+        fp8_buf,
+        scale_buf,
+        num_tokens,
         heads_per_group=heads_per_group,
         o_stride_token=o.stride(0),
         o_stride_head=o.stride(1),
@@ -220,23 +261,52 @@ def fused_inv_rope_fp8_quant(
         eps=1e-10,
         QUANT_GROUP_SIZE=quant_group_size,
         CHUNKS_PER_HEAD=chunks_per_head,
-        ROPE_START=nope_dim % quant_group_size,
-        HALF_ROPE=rope_dim // 2,
+        ROPE_START=rope_start,
+        HALF_ROPE=half_rope,
         TMA_ALIGNED_SCALES=tma_aligned_scales,
         num_stages=1,
         launch_pdl=False,
-    )
-
-    grid = (tma_aligned_T, n_groups * heads_per_group)
-    _fused_inv_rope_fp8_quant_per_head[grid](
-        o,
-        positions,
-        cos_sin_cache,
-        fp8_buf,
-        scale_buf,
-        num_tokens,
-        **common_args,
         num_warps=1,
     )
+    return fp8_buf, scale_buf
 
-    return fp8_buf.transpose(0, 1), scale_buf.transpose(0, 1)
+
+def _fused_inv_rope_fp8_quant_kernel_fake(
+    o: torch.Tensor,
+    positions: torch.Tensor,
+    cos_sin_cache: torch.Tensor,
+    heads_per_group: int,
+    quant_group_size: int,
+    chunks_per_head: int,
+    rope_start: int,
+    half_rope: int,
+    tma_aligned_scales: bool,
+    fp8_max: float,
+    tma_aligned_T: int,
+    num_tokens: int,
+    n_groups: int,
+    d: int,
+    scale_inner: int,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    fp8_buf = torch.empty(
+        (n_groups, num_tokens, d),
+        dtype=torch.float8_e4m3fn,
+        device=o.device,
+    )
+    scale_dtype = torch.int32 if tma_aligned_scales else torch.float32
+    scale_buf = torch.empty(
+        n_groups * scale_inner * tma_aligned_T,
+        dtype=scale_dtype,
+        device=o.device,
+    ).as_strided(
+        (n_groups, num_tokens, scale_inner),
+        (scale_inner * tma_aligned_T, 1, tma_aligned_T),
+    )
+    return fp8_buf, scale_buf
+
+
+direct_register_custom_op(
+    op_name="fused_inv_rope_fp8_quant_kernel",
+    op_func=_fused_inv_rope_fp8_quant_kernel_impl,
+    fake_impl=_fused_inv_rope_fp8_quant_kernel_fake,
+)

